uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/GuvnorM2Repository.java

```diff
@@ -460,22 +464,47 @@ public String getPomText(final String path) {
         }
     }
 
-    private static String loadPomFromJar(final File file) {
-        try (ZipFile zip = new ZipFile(file)) {
-            for (Enumeration e = zip.entries(); e.hasMoreElements(); ) {
-                ZipEntry entry = (ZipEntry) e.nextElement();
+    public String getKModuleText(final String path) {
+        return loadKieTextFromJar(path, META_INF, "kmodule.xml");
+    }
 
-                if (entry.getName().startsWith("META-INF/maven") && entry.getName().endsWith("pom.xml")) {
-                    return zipEntryToString(zip,
-                                            entry);
+    public String getKieDeploymentDescriptorText(final String path) {
+        return loadKieTextFromJar(path, META_INF, "kie-deployment-descriptor.xml");
+    }
+
+    protected String loadKieTextFromJar(final String jarFilePath, final String subPath, final String fileName) {
+        ArtifactRepository repository = this.getArtifactRepository(ArtifactRepositoryService.GLOBAL_M2_REPO_NAME);
+        final File file = new File(repository.getRootDir(),
+                                   jarFilePath);
+
+
+        final String normalizedPath = file.toPath().normalize().toString();
+        if (isJar(normalizedPath) || isKJar(normalizedPath)) {
+            return loadFileTextFromJar(file, subPath, fileName);
+        }
+        return "";
```

Comment:
TLDR; I would return null here.
I can see a two ways we're handling "issue with the file" .... here, we're returning empty string in case the filename does NOT end with .jar or .kjar.
If it does end with the suffixes, but the file is corrupt or error during reading happens, we return null from loadFileTextFromJar method.
Empty string should be IMO returned only when the file exists, is a jar/kjar has a corresponding zipentry and the entry is an empty file.